src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -55,122 +62,173 @@ public static SyncerFactory of(
+                      format(
+                          "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+                          trimmedValue,
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
+                  throw new IllegalArgumentException(errorMessage);
+                }
+              })
+          .collect(Collectors.toList());
+    }
+  }
+
+  private static void isSyncOptionValueAll(String syncOptionValue) {
+    if (("all").equalsIgnoreCase(syncOptionValue)) {
+      final String errorMessage =
+          format(
+              "Wrong arguments supplied to \"-%s\" or \"--%s\" option! "
+                  + "'all' option cannot be passed along with other arguments.\" %s",
+              SYNC_MODULE_OPTION_SHORT, SYNC_MODULE_OPTION_LONG, SYNC_MODULE_OPTION_DESCRIPTION);
+
+      throw new IllegalArgumentException(errorMessage);
+    }
+  }
+
+  private static void isSyncOptionValueBlank(String syncOptionValue) {
     if (isBlank(syncOptionValue)) {
       final String errorMessage =
           format(
               "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
               SYNC_MODULE_OPTION_SHORT, SYNC_MODULE_OPTION_LONG, SYNC_MODULE_OPTION_DESCRIPTION);
 
-      return exceptionallyCompletedFuture(new IllegalArgumentException(errorMessage));
+      throw new IllegalArgumentException(errorMessage);
     }
+  }
 
-    Syncer<
-            ? extends ResourceView,
-            ?,
-            ? extends BaseSyncStatistics,
-            ? extends BaseSyncOptions<?, ?>,
-            ? extends QueryDsl<?, ?>,
-            ? extends BaseSync<?, ?, ?>>
-        syncer;
+  private static Collection<List<SyncModuleOption>> groupSyncModuleOptions(
+      List<SyncModuleOption> syncModuleOptions) {
+    Map<SyncModuleOption, Integer> syncModuleOptionWithReferencesCount =
+        syncModuleOptions
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    Function.identity(),
+                    syncModuleOption ->
+                        (syncModuleOption.getReferences().isEmpty())
+                            ? 1
+                            : countReferences(
+                                syncModuleOption.getReferences(), syncModuleOptions)));
 
-    try {
-      syncer = buildSyncer(syncOptionValue, runnerNameOptionValue, isSyncProjectSyncCustomObjects);
-    } catch (IllegalArgumentException exception) {
+    Map<Integer, List<SyncModuleOption>> groupedSyncModuleOptions = new TreeMap<>();
+    for (Map.Entry<SyncModuleOption, Integer> pivotEntry :
+        syncModuleOptionWithReferencesCount.entrySet()) {
+      if (!groupedSyncModuleOptions.containsKey(pivotEntry.getValue())) {
+        List<SyncModuleOption> syncOptionsList = new ArrayList<>();
+        for (Map.Entry<SyncModuleOption, Integer> iteratingEntry :
```

Comment:
leave an empty line before a `for` loop statement line. It helps visually separate out logical blocks of code.

Reply:
done https://github.com/commercetools/commercetools-project-sync/pull/219/commits/39701b0756b605aa70a49b931d101e18bacc033f